files/fluentd/run.sh

```diff
@@ -77,7 +77,13 @@ export TOTAL_LIMIT_SIZE
 # Calculate the max number of queued chunks given the size of each chunk
 # and the max allowed space per buffer
 ##
-BUFFER_SIZE_LIMIT=$(echo ${BUFFER_SIZE_LIMIT:-8m} |  sed -e "s/[Kk]/*1024/g;s/[Mm]/*1024*1024/g;s/[Gg]/*1024*1024*1024/g;s/i//g" | bc)
+BUFFER_SIZE_RE='^[0-9]+[kKmMgG]?$'
```

Comment:
I'm missing the reason we need to execute a regex against this value.  The sed will do nothing if there is no match leaving you with the raw value.  Also we can default missing values with straight bash substitutions

Reply:
Because in fluent.conf file, if we havn't provided any value in ClusterForwarder CR then chunk_limit_size will be set to "#{ENV['BUFFER_SIZE_LIMIT'] || '8m'}". So if this is set then will go to else and see if environment variable is set. If that's also not set then default value of "8m" will be set as chunk_limit_size

Reply:
It's pretty horrible to be doing this in bash at all. Why aren't we doing this in Go using the standard https://godoc.org/k8s.io/apimachinery/pkg/api/resource Quantity to serialize/deserialize?
@sasagarw I understand you're following the existing pattern, but maybe it's time to do some clean up.